bindings/java/capstone/Capstone.java

```diff
@@ -135,17 +139,34 @@ public CsInsn (_cs_insn insn, int _arch, NativeLong _csh, CS _cs, boolean diet)
       raw = insn;
       csh = _csh;
 
+
       if (insn.cs_detail != null) {
-        if (!diet) {
-          regsRead = new short[insn.cs_detail.regs_read_count];
-          for (int i=0; i<regsRead.length; i++)
-            regsRead[i] = insn.cs_detail.regs_read[i];
-          regsWrite = new short[insn.cs_detail.regs_write_count];
-          for (int i=0; i<regsWrite.length; i++)
-            regsWrite[i] = insn.cs_detail.regs_write[i];
-          groups = new byte[insn.cs_detail.groups_count];
-          for (int i=0; i<groups.length; i++)
-            groups[i] = insn.cs_detail.groups[i];
+        if (!diet) {            
+	    	// Allocating memory for array of registers
+	        Pointer regsReadArray = new Memory(16*64); 
+	        Pointer regsWriteArray = new Memory(16*64);
+	        
+	        // Create reference for sizes
+	  	  	ByteByReference regs_read_count = new ByteByReference();
+	  	  	ByteByReference regs_write_count = new ByteByReference();
+	
+	  	  	// get information
+	        cs.cs_regs_access(_csh, insn.getPointer(), regsReadArray, regs_read_count, regsWriteArray, regs_write_count);
+	        
+	        regsRead = new short[regs_read_count.getValue()];
+	        for (int i=0; i<regsRead.length; i++)
+	        	regsRead[i] = regsReadArray.getShort(i*2);
+	        
+	        regsWrite = new short[regs_write_count.getValue()];
+	        for (int i=0; i<regsWrite.length; i++)
+	        	regsWrite[i] = regsWriteArray.getShort(i*2);
```

Comment:
those are empty because nobody did that for Java binding yet. compared to Python binding, which is very well supported, Java binding still misses many features. so while you are at it, please fix them all.